api/src/pcapi/core/offers/api.py

```diff
@@ -998,13 +1002,111 @@ def set_offer_status_based_on_fraud_criteria(offer: AnyOffer) -> models.OfferVal
     else:
         status = models.OfferValidationStatus.APPROVED
 
     logger.info("Computed offer validation", extra={"offer": offer.id, "status": status.value})
     return status
 
 
+def get_data_compliance_scoring(offer: models.Offer) -> int | None:
+    data = _get_offer_data_for_compliance_api(offer)
+
+    client_id = settings.COMPLIANCE_API_CLIENT_ID
+
+    if settings.IS_DEV:
+        # This is a fake auth_token; if you want to test it in development env
+        # you have to use the result returned by 'gcloud auth print-access-token'
+        auth_token = "gcloud auth print-access-token"
+
+        try:
+            id_token_response = requests.post(
+                f"https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/{settings.COMPLIANCE_API_SERVICE_ACCOUNT}:generateIdToken",
+                headers={
+                    "Authorization": f"Bearer {auth_token}",
+                    "Content-Type": "application/json; charset=utf-8",
+                },
+                json={"audience": client_id, "includeEmail": "true"},
+            )
+        except requests.exceptions.InvalidHeader:
+            # The auth_token has probably not been changed to a correct value
+            logger.info("Check the value of your auth_token")
+            return None
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.info("Could not get id_token", extra={"exc": exc})
+            return None
+
+        id_token = id_token_response.json()["token"]
+
+    else:
+        id_token = offers_utils.get_id_token_from_google(client_id)
+
+    try:
+        api_response = requests.post(
+            "https://compliance.passculture.team/latest/model/compliance/scoring",
+            headers={
+                "Authorization": f"Bearer {id_token}",
+                "accept": "application/json",
+                "Content-Type": "application/json",
+            },
+            json=data,
+        )
+
+        data_score = api_response.json()["probability_validated"]
+    except Exception as exc:  # pylint: disable=broad-except
+        logger.info("Connection to API Compliance failed", extra={"exc": exc})
+        return None
+
+    return data_score
+
+
+def _get_offer_data_for_compliance_api(offer: models.Offer) -> dict:
+    rayon = offer.extraData.get("rayon") if offer.extraData else None
+    macroRayon = (
+        (
+            models.BookMacroSection.query.filter(sa.func.lower(models.BookMacroSection.section) == rayon)
+            .with_entities(models.BookMacroSection.macroSection)
+            .one_or_none()
+        )
+        if rayon
+        else None
+    )
+
+    offer_type_label = None
+    if offer.extraData:
+        if "showType" in offer.extraData and offer.extraData["showType"]:
+            offer_type_label = show_types.SHOW_TYPES_LABEL_BY_CODE[int(offer.extraData["showType"])]
```

Comment:
Tu fais 3 recherches de la clé `showType` dans le dictionnaire.
Suggestion pour optimiser : 
```suggestion
        if show_type := extra_data.get("showType"):
            offer_type_label = show_types.SHOW_TYPES_LABEL_BY_CODE[int(show_type)]
```
Ça peut s'appliquer aussi aux suivants.

Reply:
Mais si le `get` renvoie `None`, alors on aura `int(None)`, non ?

Reply:
J'suis fatigué moi